packages/next/src/state.tsx

```diff
@@ -0,0 +1,153 @@
+"use client";
+
+export type Maybe<TType> = TType | null;
+import React, {
+  type PropsWithChildren,
+  createContext,
+  useContext,
+  useRef,
+} from "react";
+import { createStore, useStore } from "zustand";
+
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface SfLocales {}
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface SfCart {}
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface SfCurrency {}
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface SfCustomer {}
```

Comment:
Nice, I'm only wondering what the `.d.ts` on the Storefront side will look like?

Reply:
Unfortunately this solution is now changed to use the generics.